lib/class-wp-theme-json-gutenberg.php

```diff
@@ -841,9 +841,27 @@ protected static function sanitize( $input, $valid_block_names, $valid_element_n
 
 		$schema_styles_blocks   = array();
 		$schema_settings_blocks = array();
+
+		// Generate blocks schema without variations. They will be added later
+		// as each variation can support inner block styles which will need a
+		// schema for variation block styles that don't contain variations
+		// i.e. the variable block styles schema will be the same as the
+		// `$schema_styles_blocks` generated here.
```

Comment:
Any ideas on a better approach here would be greatly appreciated 🙏 

While writing some additional tests in bfa61f1cac2ec27f1eabca0a5fff6b335636e68f I found that block style variations wouldn't sanitize invalid properties within the variations. This requires the element and block styles schemas (without the variations sub-schema) used in the variations part of the final block styles schema.

Splitting this into a second foreach was about the neatest I could find.

Reply:
Good catch. Unfortunately I can't think of a better way to do it, either, as the variations need to have the complete schemas for the blocks and elements, otherwise there's no way to validate the styles and settings are valid for children of the variation. It's often a bummer to have to add in another `foreach`, but I'm not sure if it can be avoided easily here. At least it's only one more loop rather than anything recursive!